xcp_d/utils/filemanip.py

```diff
@@ -0,0 +1,985 @@
+
+    Uses otool on darwin, ldd on linux. Currently doesn't support windows.
+
+    """
+    command = None
+    if sys.platform == "darwin":
+        command = "otool -L `which %s`" % name
+    elif "linux" in sys.platform:
+        command = "ldd `which %s`" % name
+    else:
+        return "Platform %s not supported" % sys.platform
+
+    deps = None
+    try:
+        proc = sp.Popen(
+            command, stdout=sp.PIPE, stderr=sp.PIPE, shell=True, env=environ
+        )
+        o, e = proc.communicate()
+        deps = o.rstrip()
+    except Exception as ex:
+        deps = f"{command!r} failed"
+        fmlogger.warning(f"Could not get dependencies of {name}s. Error:\n{ex}")
+    return deps
+
+
+def canonicalize_env(env):
+    """Windows requires that environment be dicts with str as keys and values
+    This function converts any unicode entries for Windows only, returning the
+    dictionary untouched in other environments.
+
+    Parameters
+    ----------
+    env : dict
+        environment dictionary with unicode or bytes keys and values
+
+    Returns
+    -------
+    env : dict
+        Windows: environment dictionary with str keys and values
+        Other: untouched input ``env``
+    """
+    if os.name != "nt":
+        return env
+
+    out_env = {}
+    for key, val in env.items():
+        if not isinstance(key, str):
+            key = key.decode("utf-8")
+        if not isinstance(val, str):
+            val = val.decode("utf-8")
+        out_env[key] = val
+    return out_env
+
+
+def relpath(path, start=None):
+    """Return a relative version of a path"""
+
+    try:
+        return op.relpath(path, start)
+    except AttributeError:
+        pass
+
+    if start is None:
+        start = os.curdir
+    if not path:
+        raise ValueError("no path specified")
+    start_list = op.abspath(start).split(op.sep)
+    path_list = op.abspath(path).split(op.sep)
+    if start_list[0].lower() != path_list[0].lower():
+        unc_path, rest = op.splitunc(path)
+        unc_start, rest = op.splitunc(start)
+        if bool(unc_path) ^ bool(unc_start):
+            raise ValueError(
+                ("Cannot mix UNC and non-UNC paths " "(%s and %s)") % (path, start)
+            )
+        else:
+            raise ValueError(
+                "path is on drive %s, start on drive %s" % (path_list[0], start_list[0])
+            )
+    # Work out how much of the filepath is shared by start and path.
+    for i in range(min(len(start_list), len(path_list))):
+        if start_list[i].lower() != path_list[i].lower():
+            break
+    else:
+        i += 1
+
+    rel_list = [op.pardir] * (len(start_list) - i) + path_list[i:]
+    if not rel_list:
+        return os.curdir
+    return op.join(*rel_list)
+
+
+@contextlib.contextmanager
+def indirectory(path):
+    cwd = os.getcwd()
+    os.chdir(str(path))
+    try:
+        yield
+    finally:
+        os.chdir(cwd)
```

Comment:
do we need all of the other functions in here? Or just the one that supports the cifti extensions?